### PR TITLE
Improved support for public API

### PR DESCRIPTION
**Pull request by netvl**
* Added support for bypassing `json` argument to requests session
* Fixed content type configuration for public API methods

### CODE REVIEW

**Comment by netvl** [comment]
@vasichkin please review

File: qubell/api/provider/__init__.py [hunk 1]
```diff
@@ -61,16 +61,19 @@ def get_destination_url():
...
 
             #add json content type for:
-            # - all public api, meaning have basic auth
-            # - private that ends with .json
             # - unless files are sent
-            if "files" not in bypass_args and (destination_url.endswith('.json') or "auth" in route_args):
+            # - private that ends with .json
+            # - all public api with POST/PUT method, meaning have basic auth
+            # - json parameter is present
+            if "files" not in bypass_args and (destination_url.endswith('.json') or "json" in route_args or ("auth" in bypass_args and method in ["POST", "PUT"])):
```
Comment (dmakhno) [inline qubell/api/provider/__init__.py]: can you please update unit tests for this?
https://github.com/qubell/contrib-python-qubell-client/blob/master/qubell/tests/provider/test_route_decorators.py

Comment (netvl) [inline qubell/api/provider/__init__.py]: sure, didn't see them

Comment (vasichkin) [inline qubell/api/provider/__init__.py]: Agreed, could be merged after updating test.

Comment (netvl) [inline qubell/api/provider/__init__.py]: I've added some tests.

**Comment by netvl** [comment]
BTW, adding content type regardless of the HTTP method is really bad - `Content-Type` is not a valid header for `GET` or `DELETE` requests, for example, because these requests do not contain body. The server really dislikes such requests. I would have removed it entirely, I just didn't want to break something which depends on it.

**Comment by netvl** [comment]
@vasichkin bump

**Comment by netvl** [comment]
@dmakhno, @vasichkin, @chemikadze As we concluded in the meeting, I've added all routes from qubell/vermilion#6376 here. They should provide the basis for further client refactoring, as @chemikadze mentioned.

File: qubell/api/provider/router.py [hunk 2]
```diff
@@ -399,6 +402,114 @@ def post_instance_workflow(self, org_id, instance_id, wf_name, auth, data="{}"):
...
     def get_environments(self, org_id, auth): pass
 
+    # Above methods are kept for backwards compatibility; when/if the client is refactored,
+    # they should go away, as well as the extension of `PrivatePath`.
+
+    # Applications
+
+    @basic_auth
+    @route("POST /api/1/applications/{app_id}/launch")
+    def api1_application_launch(self, app_id, json, auth): pass
```
Comment (dmakhno) [inline qubell/api/provider/router.py]: ok, with me, but still think it should be part of "PrivatePath", anyway next refactoring will put these on proper place, and I don't have major objections.

**Comment by dmakhno** [comment]
no stopper objections. refactoring is a big debt.

**Comment by netvl** [comment]
@vasichkin?

**Comment by netvl** [comment]
@vasichkin, do you have any ideas why one of travis build has failed? It looks like it was launched without QUBELL_* environment variables - I got similar output when I tried to run integration tests locally. I'm not sure why it happened.

**Comment by dmakhno** [comment]
@netvl integration tests cannot be run from PR
`${TRAVIS_PULL_REQUEST} == "false"`
not sure why they are actually executed...

**Comment by netvl** [comment]
Yeah, that's what I thought as well. The job with `pr` suffix completed successfully. It is the other job that failed, with `push` suffix. I don't know what it is...

**Comment by dmakhno** [comment]
I assume you've pushed this branch to this repo, and github by commit linked them together.

**Comment by netvl** [comment]
Oh, yeah, indeed you're right. I've pushed the branch here because I need it in vermilion PR. Thanks!

**Comment by vasichkin** [comment]
No objections

**Comment by netvl** [comment]
Thanks!

**Pull request merged**